pkg/deployment/deployment.go

```diff
@@ -315,6 +315,34 @@ func (builder *Builder) WithServiceAccountName(serviceAccountName string) *Build
 	return builder
 }
 
+// WithVolume sets the Volume on the deployment definition.
+func (builder *Builder) WithVolume(volume *coreV1.Volume) *Builder {
```

Comment:
I see that this code adds a volume to a deployment but I don't see how any containers within the deployment can consume it. I think we should include another method that accepts a container name and a *coreV1.VolumeMount in order to make use of the volume. WDYT?

Reply:
@trewest Do you suggest something like that? >
https://github.com/openshift-kni/eco-goinfra/blob/8a2c85bc508cd911d35c4ac203df41ea6330b46b/pkg/pod/pod.go#L603C5-L603C5

Reply:
Yeah, the only thing is that this method is limited to configmap volume sources so it may get a bit difficult to write a method for each volume source type and all of their supported parameters (there are many variations)

Reply:
Can you please share what you propose?

Reply:
it is not so clear how you think we could how can we relate one to the other?
WE already have the method dedicated to the container configuration: WithAdditionalContainerSpecs.
What you ask me to do will add a complicity and I believe end-user has to take care about it.

Reply:
and, actually, I have no idea how we could implement such verification when two different methods involved in obj creation and object is not created yet

Reply:
@kononovn First I think we need to answer the question of do we only support mapping a volume to all containers in a pod or should the user be able to choose which containers the volume is attached to? For instance, two containers may be able to use the same volume but the mountPath in each of those containers could be different, this means we would have to decouple the WithVolume and WithVolumeMount operations. Take for instance:
```
	deployment.NewBuilder(APIClient, "test-deployment", "test-deployment", map[string]string{"app": "test-deployment"}, &v1.Container{
		Name:  "web",
		Image: "nginx",
	}).WithAdditionalContainerSpecs([]v1.Container{
		{
			Name:  "database",
			Image: "postgres",
		},
	})

	userVol := &v1.Volume{
		Name: "uservol",
		VolumeSource: v1.VolumeSource{
			ConfigMap: &v1.ConfigMapVolumeSource{
				LocalObjectReference: v1.LocalObjectReference{
					Name: "user-data-configmap",
				},
			},
		},
	}

	deployment.WithVolume(userVol).WithVolumeMount("web", userVol.Name, v1.VolumeMount{
		ReadOnly:  true,
		MountPath: "/usr/share/nginx/users",
	}).WithVolumeMount("database", userVol.Name, v1.VolumeMount{
		ReadOnly:  true,
		MountPath: "/etc/postgresql/users.conf",
	}).Create()
```

So if we wanted to support something like this, the methods should look like
```
func (builder *Builder) WithVolume(volume *coreV1.Volume) *Builder {}
func (builder *Builder) WithVolumeMount(containerName, volumeName string, volumeMount *coreV1.VolumeMount) *Builder {}
```

Reply:
@trewest Let's support  mapping a volume to all containers  for now and if later we got requirements or real use-case scenario we can change it by adding new API method. Something like WithVolumeInContainer or WithVolumeConfig. WDYT ?

Reply:
Okay if that is the case, let's do it like the pod [pkg](https://github.com/openshift-kni/eco-goinfra/blob/8a2c85bc508cd911d35c4ac203df41ea6330b46b/pkg/pod/pod.go#L603C5-L603C5) but take in a volume rather than only supporting configmap. Sound good?
```
func (builder *Builder) WithVolume(volume *coreV1.Volume, mountPath string ) *Builder {}
```
We should just verify that `volume != nil && volume.Name != ""`

Reply:
@trewest  Ok agree +1. 
@elenagerman Please refactor this function according to this example https://github.com/openshift-kni/eco-goinfra/blob/8a2c85bc508cd911d35c4ac203df41ea6330b46b/pkg/pod/pod.go#L603C5-L603C5 . Please note that you don't need to change current argument **volume *coreV1.Volume** to **volumeName**
The main idea of refactoring is to make sure that the volume is forwarded in to all containers under deployment.